tools/metadata_generation/swapi_metadata_generator.py

```diff
@@ -0,0 +1,97 @@
+"""SWAPI metadata generator."""
+
+import json
+
+import pandas as pd
+
+
+def get_global_attributes(sheet):
+    """Get the global attributes from a metadata spreadsheet.
+
+    Parameters
+    ----------
+    sheet : pandas.DataFrame
+        A sheet from an Excel file represented as a pandas DataFrame.
+
+    Returns
+    -------
+    dict
+        A dictionary of the global attributes, where the keys are the attribute names
+        and the values are the attribute descriptions.
+    """
+    sheet = sheet.set_index("NAME")
+    # subset by global_att field type, then only keep the CATDESC column
+    sheet = sheet[sheet["FIELD_TYPE"] == "global_att"][["CATDESC"]]
+    return sheet.to_dict()["CATDESC"]
+
+
+def get_file_attributes(sheet, global_attrs=None):
+    """Get the file attributes from a metadata spreadsheet.
+
+    Parameters
+    ----------
+    sheet : pandas.DataFrame
+        A sheet from an Excel file represented as a pandas DataFrame.
+
+    Returns
+    -------
+    dict
+        A dictionary of the file attributes, where the keys are the attribute names
+        and the values are the attribute descriptions.
+    """
+    # Add the global attributes for this specific sheet
+    file_attrs = {"global_attrs": get_global_attributes(sheet)}
+    # Add the global attributes from the global_attrs sheet (if any)
+    file_attrs["global_attrs"].update(global_attrs or {})
+
+    # Move onto variable and coordinate metadata
+    # Get everything that isn't a global attribute and drop that column
+    sheet = sheet[sheet["FIELD_TYPE"] != "global_att"].drop(columns=["FIELD_TYPE"])
+    sheet = sheet.set_index("NAME")
```

Comment:
I think I understand what this does. But to double check, does this `drop` call filters out all non-global attrs and gets those columns and assign it to `sheet`? It may be helpful to have different name than `sheet`. I suggested something but you are better with names. Can you pick another name?
```suggestion
    data_or_coord_data = sheet[sheet["FIELD_TYPE"] != "global_att"].drop(columns=["FIELD_TYPE"])
    data_or_coord_data = data_or_coord_data.set_index("NAME")
```

Reply:
This is modifying the structure/format of "sheet". I would call it `df` or `variable_data` if anything else I guess since it is a pandas dataframe. It is pretty standard in pandas to overwrite your local `df` variable as you modify it, so I'm not sure how helpful renaming would be though.

There is a comment above this saying we are getting all rows that aren't equal to `global_att` as our subset. Then we are getting rid of that column because we don't want to propagate it in the output metadata. Or do we need it in the CDF metadata as well?

Reply:
Updated to `variables`

Reply:
I was thinking `variable_or_coord_data` when I made suggestions but wrote something else. `variables` sounds good.

No, we don't need that column after that filter.